pkg/operator/storagelocations.go

```diff
@@ -70,6 +70,53 @@ func (op *defaultReportingOperator) syncStorageLocation(logger log.FieldLogger,
 	return nil
 }
 
+// UpdateStorageLocation is responsible for determining if the StorageLocation
+// the Reporting Operator is currently processing needs to be updated in
+// some capacity and performing the manipulations of those fields. Note:
+// the actual call to the k8s client to update this resource needs to be
+// made at the call site of this method.
+func (op *defaultReportingOperator) UpdateStorageLocation(logger log.FieldLogger, storageLocation *metering.StorageLocation) (bool, error) {
+	if storageLocation.Spec.Hive == nil {
+		return false, fmt.Errorf("only Hive storage is currently supported")
+	}
+	if storageLocation.Spec.Hive.DatabaseName == "" {
+		return false, fmt.Errorf("spec.hive.databaseName is required if spec.hive is set")
+	}
+	if !reportingutil.IsValidSQLIdentifier(storageLocation.Spec.Hive.DatabaseName) {
+		return false, fmt.Errorf("spec.hive.databaseName %s is invalid, must contain only alpha numeric values, underscores, and start with a letter or underscore", storageLocation.Spec.Hive.DatabaseName)
+	}
+
+	if storageLocation.Spec.Hive.UnmanagedDatabase {
+		logger.Infof("StorageLocation %s is unmanaged", storageLocation.Name)
+
+		if storageLocation.Status.Hive.DatabaseName != storageLocation.Spec.Hive.DatabaseName {
+			storageLocation.Status.Hive.DatabaseName = storageLocation.Spec.Hive.DatabaseName
+			return true, nil
+		}
+		if storageLocation.Status.Hive.Location != storageLocation.Spec.Hive.Location {
+			storageLocation.Status.Hive.Location = storageLocation.Spec.Hive.Location
+			return true, nil
+		}
+	}
+	if storageLocation.Status.Hive.DatabaseName == "" {
```

Comment:
I think this can go. It's already checked above, right?

Reply:
That was checking the `.Spec.Hive.DatabaseName`.